src/data/pokemon-forms.ts

```diff
@@ -240,43 +262,41 @@ export class SpeciesFormChangeRevertWeatherFormTrigger extends SpeciesFormChange
    * @returns `true` if the Pokemon will revert to its original form, `false` otherwise
    */
   override canChange(pokemon: Pokemon): boolean {
-    if (pokemon.hasAbility(this.ability, false, true)) {
-      const isWeatherSuppressed = globalScene.arena.weather?.isEffectSuppressed();
-      const isAbilitySuppressed = pokemon.summonData.abilitySuppressed;
-      const summonDataAbility = pokemon.summonData.ability;
-      const isAbilityChanged = summonDataAbility !== this.ability && summonDataAbility !== AbilityId.NONE;
+    if (!pokemon.hasAbility(this.ability, false, true)) {
+      return false;
+    }
 
-      if (
-        globalScene.arena.hasWeather(this.weathers)
-        || isWeatherSuppressed
-        || isAbilitySuppressed
-        || isAbilityChanged
-      ) {
-        return true;
-      }
+    const isWeatherSuppressed = globalScene.arena.weather?.isEffectSuppressed();
+    const isAbilitySuppressed = pokemon.summonData.abilitySuppressed;
+    const summonDataAbility = pokemon.summonData.ability;
+    const isAbilityChanged = summonDataAbility !== this.ability && summonDataAbility !== AbilityId.NONE;
+
+    if (globalScene.arena.hasWeather(this.weathers) || isWeatherSuppressed || isAbilitySuppressed || isAbilityChanged) {
+      return true;
     }
+
     return false;
```

Comment:
```suggestion
    return globalScene.arena.hasWeather(this.weathers)
      || isWeatherSuppressed
      || isAbilitySuppressed
      || isAbilityChanged;
```
You might have to run this through Prettier...